config/routes.rb

```diff
@@ -1,16 +1,20 @@
 Rails.application.routes.draw do
   get 'sessions/new'
   root 'main_page#landing'
-  get 'main_page/home'
-  get 'main_page/landing'
-  resources :products
+  resources :products, only: %i[index new show edit create update destroy] do
+    post :add_to_cart, on: :member
+    delete :remove_single_item_from_cart, on: :member
+    delete :remove_product_from_cart, on: :member
+  end
   resources :users
```

Comment:
Please add an empty line before resources :users and group it with get '/signup', to: 'users#new'